### ISSUE TITLE
Fix `declaration-block-no-shorthand-property-overrides` false-negative with `background`

### ISSUE THREAD

**Issue by silverwind** [stylelint/stylelint#8349]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
  background: red;
  background-color: red;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-no-shorthand-property-overrides" : true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAjJYA1gOYBOEArplPCqTFANxY75FmXUC0kANhKVr0mWAL4gANCABmAS14wAckgC2cRDAAeqgA4KAdGADORyeAiY5xBCAzYU6EKQoKjj2nZw5HsMLySkSAAushZcuPxEXJgQXEYAFgJB8UjcOuQ6MKRBAJ5cEABuWaSysG4gKLRBzjAsKKJiZpCWssQAYgIqwTYAVkYWZrA6poieDiBGuQq8sphB7uP+QTCTjhJ1jpM507NBPBZWcUGpUAFQC45LK-MgjaJAA

### Which Stylelint-related dependencies are you using?

stylelint@16.13.2

### What did you expect to happen?

Problem to be flagged because `background` is a shorthand for `background-color`.

### What actually happened?

No errors being reported.

### Do you have a proposal to fix the bug?

I think it is safe to treat `background` and `background-color` as overriding each other, at least with a single argument to `background`.

**Comment by a2937** [comment]
I can confirm this issue. Silly me thought this was an issue with the config I was using.

**Comment by jeddy3** [comment]
@silverwind Thanks for the report and for using the template.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
I jumped the gun there...

The rule checks shorthand overriding longhand:

```css
a {
  background-color: red;
  background: red;
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAjJYA1gOYBOEArplALSQA2Ep8KZMMmA3FjvkWZWotSMKN0wBfEABoQAMwCW9GADkkAWziIYADw0AHZQDowAZ1MzwETIuIIQGbCnQhSFZaZctHOHC9hg9EikSAAuCtY0uIxENJgQNKYAFkyhSUjUNPrk+jCkoQCeNBAAbnmkCrCeICgsoW4wPCgSWFKykDYKxABiTOph9gBWptaWsPoWiD7OIKaFyvQKmKFeM0GhMHMu0k0ucwULS6F01raJoRlQwVCrLuubKyCtIBJAA).

We don't currently have a rule for checking:

```css
a {
  background: red;
  background-color: red;
}
```

It seems using a shorthand property first, and then overriding specific properties is a typical pattern. 

Can we think of more use cases in this area or specific problematic patterns we'll want to check for?

**Comment by Mouvedia** [comment]
It looks like a request for a new rule that would belong in the [redundant category](https://stylelint.io/user-guide/rules/#redundant).
i.e. `declaration-block-no-redundant-shorthand-properties` would be the pendant of the existing [`declaration-block-no-redundant-longhand-properties`](https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties).

**Comment by jeddy3** [comment]
We can treat this as a false negative of [declaration-block-no-redundant-longhand-properties](https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties). In the example, ` background-color: red;` is a redundant longhand property because the value is already being set in a previous shorthand property. The same is true for things like:

```css
a {
  margin: 10px;
  margin-left: 10px;
}
```

@silverwind I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time. The fix can just target single values that are the same.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ryo-manba** [comment]
Hi @jeddy3, I have a question about the expected behavior in this issue.

I’m wondering whether detecting if the values are *exactly the same* might be more difficult than it seems.

For example:

```css
a {
  margin: 10px 20px;
  margin-left: 20px;
}
```

In this case, `margin-left: 20px` is redundant, but only if we correctly parse the shorthand and determine that `left = 20px`.

Similarly, consider this background case:

```css
a {
  background: url("foo.png") 40% / 10em red;
  background-color: red;
}
```

Here as well, `background-color: red` is redundant, but we’d need to extract the color from the shorthand value.

In this fix, are we aiming to support such complex cases where we parse the shorthand syntax to determine redundancy?  
Or is it intended to cover only simpler cases (e.g. `background: red; background-color: red;`) where the values are exactly the same as strings?

**Comment by jeddy3** [comment]
@ryo-manba Thank you for the clearly written question. 

I think we can limit the first PR to addressing simpler cases, as the effort needed to fix the complex cases may not be worth it. 

I suggest we start with matching strings. If there are relatively simple extractions, e.g. edges of a side, we can consider those for a follow-on issue/PR as that'll cover some of the more common use cases like `margin` and `padding`.

**Comment by CwPranay** [comment]
Hi, I’d like to work on this issue. Can I take it?

**Comment by jeddy3** [comment]
Closed by https://github.com/stylelint/stylelint/pull/8892
